metadata.rb

```diff
@@ -9,6 +9,7 @@
 depends          'sudo'
 depends          'java'
 depends          'apache2'
+depends          'yum', '~> 3.10.0'
```

Comment:
Without this, the RHEL installation does not work. Should it really be up to a higher level resource? (Chef Role, for example) to include that? Your recipe for RHEL installation uses a resource from the yum cookbook. Therefore, your cookbook has a hard dependency on the yum cookbook.